azconfig/client/src/main/java/com/azure/azconfig/RequestIdPolicy.java

```diff
@@ -0,0 +1,32 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.azconfig;
+
+import com.microsoft.rest.v3.http.HttpPipelineCallContext;
+import com.microsoft.rest.v3.http.HttpResponse;
+import com.microsoft.rest.v3.http.NextPolicy;
+import com.microsoft.rest.v3.http.policy.HttpPipelinePolicy;
+import reactor.core.publisher.Mono;
+
+import java.util.UUID;
+
+/**
+ * Creates a policy which puts a UUID in the request header. Azure uses
+ * the request id as the unique identifier for the request.
+ * Also sets 'x-ms-return-client-request-id' header to tell server to return request id in the response.
+ */
+public class RequestIdPolicy implements HttpPipelinePolicy {
+    private static final String REQUEST_ID_HEADER = "x-ms-client-request-id";
+    private static final String ECHO_REQUEST_ID_HEADER = "x-ms-return-client-request-id";
+
+    @Override
+    public Mono<HttpResponse> process(HttpPipelineCallContext context, NextPolicy next) {
+        String requestId = context.httpRequest().headers().value(REQUEST_ID_HEADER);
+        if (requestId == null) {
+            context.httpRequest().headers().set(REQUEST_ID_HEADER, UUID.randomUUID().toString());
+        }
+        context.httpRequest().headers().set(ECHO_REQUEST_ID_HEADER, "true");
+        return next.process();
+    }
+}
```

Comment:
I'm not sure this needs to be public API. The justification that it might be required to be public API is so that other developers can create a custom HTTP pipeline and still be able to connect. In that case, we should document the relevance of this class, why it exists, how to use it, where it should be used, etc. When a client has specific pipeline policies, it too should document the need for these policies to properly connect to the service.